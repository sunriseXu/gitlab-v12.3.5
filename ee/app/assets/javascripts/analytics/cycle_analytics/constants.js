import { __ } from '~/locale';

export const DEFAULT_DATA_TIME_FRAME = 30;

export const EVENTS_LIST_ITEM_LIMIT = 50;

export const EMPTY_STAGE_TEXT = {
  issue: __(
    'The issue stage shows the time it takes from creating an issue to assigning the issue to a milestone, or add the issue to a list on your Issue Board. Begin creating issues to see data for this stage.',
  ),
  plan: __(
    'The planning stage shows the time from the previous step to pushing your first commit. This time will be added automatically once you push your first commit.',
  ),
  code: __(
    'The coding stage shows the time from the first commit to creating the merge request. The data will automatically be added here once you create your first merge request.',
  ),
  test: __(
    'The testing stage shows the time GitLab CI takes to run every pipeline for the related merge request. The data will automatically be added after your first pipeline finishes running.',
  ),
  review: __(
    'The review stage shows the time from creating the merge request to merging it. The data will automatically be added after you merge your first merge request.',
  ),
  staging: __(
    'The staging stage shows the time between merging the MR and deploying code to the production environment. The data will be automatically added once you deploy to production for the first time.',
  ),
  production: __(
    'The production stage shows the total time it takes between creating an issue and deploying the code to production. The data will be automatically added once you have completed the full idea to production cycle.',
  ),
};
