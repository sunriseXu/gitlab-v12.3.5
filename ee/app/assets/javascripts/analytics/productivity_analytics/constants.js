import { __, s__ } from '~/locale';

export const chartKeys = {
  main: 'main',
  timeBasedHistogram: 'timeBasedHistogram',
  commitBasedHistogram: 'commitBasedHistogram',
  scatterplot: 'scatterplot',
};

export const chartTypes = {
  histogram: 'histogram',
  scatterplot: 'scatterplot',
};

export const metricTypes = [
  {
    key: 'time_to_first_comment',
    label: __('Time from first commit until first comment'),
    chart: chartKeys.timeBasedHistogram,
  },
  {
    key: 'time_to_last_commit',
    label: __('Time from first comment to last commit'),
    chart: chartKeys.timeBasedHistogram,
  },
  {
    key: 'time_to_merge',
    label: __('Time from last commit to merge'),
    chart: chartKeys.timeBasedHistogram,
  },
  {
    key: 'commits_count',
    label: __('Number of commits per MR'),
    chart: chartKeys.commitBasedHistogram,
  },
  {
    key: 'loc_per_commit',
    label: __('Number of LOCs per commit'),
    chart: chartKeys.commitBasedHistogram,
  },
  {
    key: 'files_touched',
    label: __('Number of files touched'),
    chart: chartKeys.commitBasedHistogram,
  },
];

export const tableSortOrder = {
  asc: {
    title: s__('ProductivityAnalytics|Ascending'),
    value: 'asc',
    icon: 'sort-lowest',
  },
  desc: {
    title: s__('ProductivityAnalytics|Descending'),
    value: 'desc',
    icon: 'sort-highest',
  },
};

export const daysToMergeMetric = {
  key: 'days_to_merge',
  label: s__('ProductivityAnalytics|Days to merge'),
};

export const defaultMaxColumnChartItemsPerPage = 20;

export const maxColumnChartItemsPerPage = {
  [chartKeys.main]: 40,
};

export const dataZoomOptions = [
  {
    type: 'slider',
    bottom: 10,
    start: 0,
  },
  {
    type: 'inside',
    start: 0,
  },
];

/**
 * #418cd8 --> $blue-400 (see variables.scss)
 */
export const columnHighlightStyle = { color: '#418cd8', opacity: 0.8 };

export const accessLevelReporter = 20;

export const projectsPerPage = 50;
