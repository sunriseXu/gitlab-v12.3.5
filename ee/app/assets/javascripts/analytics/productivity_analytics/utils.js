export const getLabelsEndpoint = (namespacePath, projectPathWithNamespace) => {
  if (projectPathWithNamespace) {
    return `/${projectPathWithNamespace}/-/labels`;
  }

  return `/groups/${namespacePath}/-/labels`;
};

export const getMilestonesEndpoint = (namespacePath, projectPathWithNamespace) => {
  if (projectPathWithNamespace) {
    return `/${projectPathWithNamespace}/-/milestones`;
  }

  return `/groups/${namespacePath}/-/milestones`;
};
