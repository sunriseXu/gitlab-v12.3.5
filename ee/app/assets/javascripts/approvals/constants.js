export const TYPE_USER = 'user';
export const TYPE_GROUP = 'group';
export const TYPE_HIDDEN_GROUPS = 'hidden_groups';

export const RULE_TYPE_FALLBACK = 'fallback';
export const RULE_TYPE_REGULAR = 'regular';
export const RULE_TYPE_CODE_OWNER = 'code_owner';
