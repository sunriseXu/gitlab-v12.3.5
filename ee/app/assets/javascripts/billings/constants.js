export const CUSTOMER_PORTAL_URL = 'https://customers.gitlab.com/subscriptions';
export const TABLE_TYPE_DEFAULT = 'default';
export const TABLE_TYPE_FREE = 'free';
export const TABLE_TYPE_TRIAL = 'trial';
