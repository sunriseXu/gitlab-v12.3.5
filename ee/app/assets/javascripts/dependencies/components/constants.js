export const WARNING = 'warning';
export const DANGER = 'danger';

export const WARNING_ALERT_CLASS = 'warning_message';
export const DANGER_ALERT_CLASS = 'danger_message';

export const WARNING_TEXT_CLASS = 'text-warning-900';
export const DANGER_TEXT_CLASS = 'text-danger-900';

// Limit the number of vulnerabilities to display so as to avoid jank.
// In practice, this limit will probably never be reached, since the
// largest number of vulnerabilities we've seen one dependency have is 20.
export const MAX_DISPLAYED_VULNERABILITIES_PER_DEPENDENCY = 50;
