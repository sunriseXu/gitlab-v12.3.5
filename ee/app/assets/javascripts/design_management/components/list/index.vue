<script>
import Design from './item.vue';

export default {
  components: {
    Design,
  },
  props: {
    designs: {
      type: Array,
      required: true,
    },
  },
};
</script>

<template>
  <ol class="list-unstyled row">
    <li v-for="design in designs" :key="design.id" class="col-md-6 col-lg-4 mb-3">
      <design
        :id="design.id"
        :event="design.event"
        :notes-count="design.notesCount"
        :image="design.image"
        :name="design.filename"
        :updated-at="design.updatedAt"
      />
    </li>
  </ol>
</template>
