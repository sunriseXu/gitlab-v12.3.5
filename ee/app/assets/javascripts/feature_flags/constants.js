import _ from 'underscore';

export const ROLLOUT_STRATEGY_ALL_USERS = 'default';
export const ROLLOUT_STRATEGY_PERCENT_ROLLOUT = 'gradualRolloutUserId';
export const ROLLOUT_STRATEGY_USER_ID = 'userWithId';

export const PERCENT_ROLLOUT_GROUP_ID = 'default';

export const DEFAULT_PERCENT_ROLLOUT = '100';

export const ALL_ENVIRONMENTS_NAME = '*';

export const INTERNAL_ID_PREFIX = 'internal_';

export const fetchPercentageParams = _.property(['parameters', 'percentage']);
export const fetchUserIdParams = _.property(['parameters', 'userIds']);
