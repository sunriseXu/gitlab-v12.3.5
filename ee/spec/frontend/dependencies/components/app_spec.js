import { GlBadge, GlEmptyState, GlLoadingIcon, GlTab } from '@gitlab/ui';
import { createLocalVue, shallowMount, mount } from '@vue/test-utils';
import { TEST_HOST } from 'helpers/test_constants';
import createStore from 'ee/dependencies/store';
import { addListType } from 'ee/dependencies/store/utils';
import { DEPENDENCY_LIST_TYPES } from 'ee/dependencies/store/constants';
import { REPORT_STATUS } from 'ee/dependencies/store/modules/list/constants';
import DependenciesApp from 'ee/dependencies/components/app.vue';
import DependenciesActions from 'ee/dependencies/components/dependencies_actions.vue';
import DependencyListIncompleteAlert from 'ee/dependencies/components/dependency_list_incomplete_alert.vue';
import DependencyListJobFailedAlert from 'ee/dependencies/components/dependency_list_job_failed_alert.vue';
import PaginatedDependenciesTable from 'ee/dependencies/components/paginated_dependencies_table.vue';

describe('DependenciesApp component', () => {
  let store;
  let wrapper;
  const { namespace } = DEPENDENCY_LIST_TYPES.all;

  const basicAppProps = {
    endpoint: '/foo',
    emptyStateSvgPath: '/bar.svg',
    documentationPath: TEST_HOST,
  };

  const factory = (props = basicAppProps) => {
    const localVue = createLocalVue();

    store = createStore();
    jest.spyOn(store, 'dispatch').mockImplementation();

    wrapper = shallowMount(DependenciesApp, {
      localVue,
      store,
      sync: false,
      propsData: { ...props },
    });
  };

  const expectComponentWithProps = (Component, props = {}) => {
    const componentWrapper = wrapper.find(Component);
    expect(componentWrapper.isVisible()).toBe(true);
    expect(componentWrapper.props()).toEqual(expect.objectContaining(props));
  };

  afterEach(() => {
    wrapper.destroy();
  });

  describe('on creation', () => {
    let dependencies;

    beforeEach(() => {
      factory();
    });

    it('dispatches the correct initial actions', () => {
      expect(store.dispatch.mock.calls).toEqual([
        ['setDependenciesEndpoint', basicAppProps.endpoint],
        ['fetchDependencies'],
      ]);
    });

    it('matches the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    describe('given a list of dependencies and ok report', () => {
      beforeEach(() => {
        dependencies = ['foo', 'bar'];

        Object.assign(store.state[namespace], {
          initialized: true,
          isLoading: false,
          dependencies,
        });
        store.state[namespace].pageInfo.total = 100;
        store.state[namespace].reportInfo.status = REPORT_STATUS.ok;

        return wrapper.vm.$nextTick();
      });

      it('matches the snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
      });

      it('passes the correct props to the paginated dependencies table', () => {
        expectComponentWithProps(PaginatedDependenciesTable, {
          namespace,
        });
      });
    });

    describe('given the dependency list job has not yet run', () => {
      beforeEach(() => {
        dependencies = [];

        Object.assign(store.state[namespace], {
          initialized: true,
          isLoading: false,
          dependencies,
        });
        store.state[namespace].pageInfo.total = 0;
        store.state[namespace].reportInfo.status = REPORT_STATUS.jobNotSetUp;

        return wrapper.vm.$nextTick();
      });

      it('matches the snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
      });
    });

    describe('given the dependency list job failed', () => {
      beforeEach(() => {
        dependencies = [];

        Object.assign(store.state[namespace], {
          initialized: true,
          isLoading: false,
          dependencies,
        });
        store.state[namespace].pageInfo.total = 0;
        store.state[namespace].reportInfo.status = REPORT_STATUS.jobFailed;
        store.state[namespace].reportInfo.jobPath = '/jobs/foo/321';

        return wrapper.vm.$nextTick();
      });

      it('matches the snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
      });

      it('passes the correct props to the job failure alert', () => {
        expectComponentWithProps(DependencyListJobFailedAlert, {
          jobPath: store.state[namespace].reportInfo.jobPath,
        });
      });

      it('passes the correct props to the paginated dependencies table', () => {
        expectComponentWithProps(PaginatedDependenciesTable, {
          namespace,
        });
      });

      describe('when the job failure alert emits the close event', () => {
        beforeEach(() => {
          const alertWrapper = wrapper.find(DependencyListJobFailedAlert);
          alertWrapper.vm.$emit('close');
          return wrapper.vm.$nextTick();
        });

        it('does not render the job failure alert', () => {
          expect(wrapper.find(DependencyListJobFailedAlert).exists()).toBe(false);
        });
      });
    });

    describe('given a dependency list which is known to be incomplete', () => {
      beforeEach(() => {
        dependencies = ['foo', 'bar'];

        Object.assign(store.state[namespace], {
          initialized: true,
          isLoading: false,
          dependencies,
        });
        store.state[namespace].pageInfo.total = 100;
        store.state[namespace].reportInfo.status = REPORT_STATUS.incomplete;

        return wrapper.vm.$nextTick();
      });

      it('matches the snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
      });

      it('passes the correct props to the incomplete-list alert', () => {
        const alert = wrapper.find(DependencyListIncompleteAlert);
        expect(alert.isVisible()).toBe(true);
      });

      it('passes the correct props to the paginated dependencies table', () => {
        expectComponentWithProps(PaginatedDependenciesTable, {
          namespace,
        });
      });

      describe('when the incomplete-list alert emits the close event', () => {
        beforeEach(() => {
          const alertWrapper = wrapper.find(DependencyListIncompleteAlert);
          alertWrapper.vm.$emit('close');
          return wrapper.vm.$nextTick();
        });

        it('does not render the incomplete-list alert', () => {
          expect(wrapper.find(DependencyListIncompleteAlert).exists()).toBe(false);
        });
      });
    });

    describe('given a fetch error', () => {
      beforeEach(() => {
        dependencies = [];

        Object.assign(store.state[namespace], {
          initialized: true,
          isLoading: false,
          errorLoading: true,
          dependencies,
        });

        return wrapper.vm.$nextTick();
      });

      it('matches the snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
      });

      it('passes the correct props to the paginated dependencies table', () => {
        expectComponentWithProps(PaginatedDependenciesTable, {
          namespace,
        });
      });
    });
  });
});

describe('DependenciesApp component with dependencyListVulnerabilities feature flag enabled', () => {
  let store;
  let wrapper;
  const { namespace: allNamespace } = DEPENDENCY_LIST_TYPES.all;
  const { namespace: vulnerableNamespace } = DEPENDENCY_LIST_TYPES.vulnerable;

  const basicAppProps = {
    endpoint: '/foo',
    emptyStateSvgPath: '/bar.svg',
    documentationPath: TEST_HOST,
  };

  const factory = (props = basicAppProps) => {
    const localVue = createLocalVue();

    store = createStore();
    addListType(store, DEPENDENCY_LIST_TYPES.vulnerable);
    jest.spyOn(store, 'dispatch').mockImplementation();

    const canBeStubbed = component => !['GlTab', 'GlTabs'].includes(component);
    const stubs = Object.keys(DependenciesApp.components).filter(canBeStubbed);

    wrapper = mount(DependenciesApp, {
      localVue,
      store,
      sync: false,
      propsData: { ...props },
      provide: {
        dependencyListVulnerabilities: true,
      },
      stubs,
    });
  };

  const setStateJobNotRun = () => {
    Object.assign(store.state[allNamespace], {
      initialized: true,
      isLoading: false,
      dependencies: [],
    });
    store.state[allNamespace].pageInfo.total = 0;
    store.state[allNamespace].reportInfo.status = REPORT_STATUS.jobNotSetUp;
  };

  const setStateLoaded = () => {
    [allNamespace, vulnerableNamespace].forEach((namespace, i, { length }) => {
      const total = length - i;
      Object.assign(store.state[namespace], {
        initialized: true,
        isLoading: false,
        dependencies: Array(total)
          .fill(null)
          .map((_, id) => ({ id })),
      });
      store.state[namespace].pageInfo.total = total;
      store.state[namespace].reportInfo.status = REPORT_STATUS.ok;
    });
  };

  const setStateJobFailed = () => {
    Object.assign(store.state[allNamespace], {
      initialized: true,
      isLoading: false,
      dependencies: [],
    });
    store.state[allNamespace].pageInfo.total = 0;
    store.state[allNamespace].reportInfo.status = REPORT_STATUS.jobFailed;
    store.state[allNamespace].reportInfo.jobPath = '/jobs/foo/321';
  };

  const setStateListIncomplete = () => {
    Object.assign(store.state[allNamespace], {
      initialized: true,
      isLoading: false,
      dependencies: [{ id: 0 }],
    });
    store.state[allNamespace].pageInfo.total = 1;
    store.state[allNamespace].reportInfo.status = REPORT_STATUS.incomplete;
  };

  const findJobFailedAlert = () => wrapper.find(DependencyListJobFailedAlert);
  const findIncompleteListAlert = () => wrapper.find(DependencyListIncompleteAlert);
  const findDependenciesTables = () => wrapper.findAll(PaginatedDependenciesTable);
  const findTabControls = () => wrapper.findAll('.gl-tab-nav-item');
  const findVulnerableTabControl = () => findTabControls().at(1);
  const findVulnerableTabComponent = () => wrapper.findAll(GlTab).at(1);

  const expectComponentWithProps = (Component, props = {}) => {
    const componentWrapper = wrapper.find(Component);
    expect(componentWrapper.isVisible()).toBe(true);
    expect(componentWrapper.props()).toEqual(expect.objectContaining(props));
  };

  const expectNoDependenciesTables = () => expect(findDependenciesTables()).toHaveLength(0);

  const expectDependenciesTables = () => {
    const { wrappers } = findDependenciesTables();
    expect(wrappers).toHaveLength(2);
    expect(wrappers[0].props()).toEqual({ namespace: allNamespace });
    expect(wrappers[1].props()).toEqual({ namespace: vulnerableNamespace });
  };

  afterEach(() => {
    wrapper.destroy();
  });

  describe('on creation', () => {
    beforeEach(() => {
      factory();
    });

    it('dispatches the correct initial actions', () => {
      expect(store.dispatch.mock.calls).toEqual([
        ['setDependenciesEndpoint', basicAppProps.endpoint],
        ['fetchDependencies'],
      ]);
    });

    it('shows only the loading icon', () => {
      expectComponentWithProps(GlLoadingIcon);
      expectNoDependenciesTables();
    });

    describe('given the dependency list job has not yet run', () => {
      beforeEach(() => {
        setStateJobNotRun();

        return wrapper.vm.$nextTick();
      });

      it('shows only the empty state', () => {
        expectComponentWithProps(GlEmptyState, { svgPath: basicAppProps.emptyStateSvgPath });
        expectNoDependenciesTables();
      });
    });

    describe('given a list of dependencies and ok report', () => {
      beforeEach(() => {
        setStateLoaded();

        return wrapper.vm.$nextTick();
      });

      it('shows both dependencies tables with the correct props', () => {
        expectDependenciesTables();
      });

      it('displays the tabs correctly', () => {
        const expected = [
          {
            text: 'All',
            total: '2',
          },
          {
            text: 'Vulnerable',
            total: '1',
          },
        ];

        const tabs = findTabControls();
        expected.forEach(({ text, total }, i) => {
          const tab = tabs.at(i);
          expect(tab.text()).toEqual(expect.stringContaining(text));
          expect(
            tab
              .find(GlBadge)
              .text()
              .trim(),
          ).toEqual(total);
        });
      });

      it('passes the correct namespace to dependencies actions component', () => {
        expectComponentWithProps(DependenciesActions, { namespace: allNamespace });
      });

      describe('given the user clicks on the vulnerable tab', () => {
        beforeEach(() => {
          findVulnerableTabControl().trigger('click');

          return wrapper.vm.$nextTick();
        });

        it('changes the current list', () => {
          expect(store.dispatch).toHaveBeenCalledWith('setCurrentList', vulnerableNamespace);
        });
      });

      describe('given the current list is the vulnerable dependencies list', () => {
        const namespace = vulnerableNamespace;
        beforeEach(() => {
          store.state.currentList = namespace;

          return wrapper.vm.$nextTick();
        });

        it('passes the correct namespace to dependencies actions component', () => {
          expectComponentWithProps(DependenciesActions, { namespace });
        });
      });

      it('has enabled vulnerable tab', () => {
        expect(findVulnerableTabComponent().classes('disabled')).toBe(false);
      });

      describe('given there are no vulnerable dependencies', () => {
        beforeEach(() => {
          store.state[vulnerableNamespace].dependencies = [];
          store.state[vulnerableNamespace].pageInfo.total = 0;

          return wrapper.vm.$nextTick();
        });

        it('disables the vulnerable tab', () => {
          expect(findVulnerableTabComponent().classes('disabled')).toBe(true);
        });
      });
    });

    describe('given the dependency list job failed', () => {
      beforeEach(() => {
        setStateJobFailed();

        return wrapper.vm.$nextTick();
      });

      it('passes the correct props to the job failure alert', () => {
        expectComponentWithProps(DependencyListJobFailedAlert, {
          jobPath: '/jobs/foo/321',
        });
      });

      it('shows both dependencies tables with the correct props', expectDependenciesTables);

      describe('when the job failure alert emits the close event', () => {
        beforeEach(() => {
          const alertWrapper = findJobFailedAlert();
          alertWrapper.vm.$emit('close');
          return wrapper.vm.$nextTick();
        });

        it('does not render the job failure alert', () => {
          expect(findJobFailedAlert().exists()).toBe(false);
        });
      });
    });

    describe('given a dependency list which is known to be incomplete', () => {
      beforeEach(() => {
        setStateListIncomplete();

        return wrapper.vm.$nextTick();
      });

      it('passes the correct props to the incomplete-list alert', () => {
        expectComponentWithProps(DependencyListIncompleteAlert);
      });

      it('shows both dependencies tables with the correct props', expectDependenciesTables);

      describe('when the incomplete-list alert emits the close event', () => {
        beforeEach(() => {
          const alertWrapper = findIncompleteListAlert();
          alertWrapper.vm.$emit('close');
          return wrapper.vm.$nextTick();
        });

        it('does not render the incomplete-list alert', () => {
          expect(findIncompleteListAlert().exists()).toBe(false);
        });
      });
    });
  });
});
