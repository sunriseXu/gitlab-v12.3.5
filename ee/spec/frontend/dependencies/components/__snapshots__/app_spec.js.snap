// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DependenciesApp component on creation given a dependency list which is known to be incomplete matches the snapshot 1`] = `
<div>
  <dependencylistincompletealert-stub />
   
  <!---->
   
  <div
    class="d-sm-flex justify-content-between align-items-baseline my-2"
  >
    <h3
      class="h5"
    >
      
        Dependencies
        
      <glbadge-stub
        data-qa-selector="dependency_list_total_count"
        pill=""
      >
        100
      </glbadge-stub>
    </h3>
     
    <dependenciesactions-stub
      namespace="allDependencies"
    />
  </div>
   
  <paginateddependenciestable-stub
    namespace="allDependencies"
  />
</div>
`;

exports[`DependenciesApp component on creation given a fetch error matches the snapshot 1`] = `
<div>
  <!---->
   
  <!---->
   
  <div
    class="d-sm-flex justify-content-between align-items-baseline my-2"
  >
    <h3
      class="h5"
    >
      
        Dependencies
        
      <!---->
    </h3>
     
    <dependenciesactions-stub
      namespace="allDependencies"
    />
  </div>
   
  <paginateddependenciestable-stub
    namespace="allDependencies"
  />
</div>
`;

exports[`DependenciesApp component on creation given a list of dependencies and ok report matches the snapshot 1`] = `
<div>
  <!---->
   
  <!---->
   
  <div
    class="d-sm-flex justify-content-between align-items-baseline my-2"
  >
    <h3
      class="h5"
    >
      
        Dependencies
        
      <glbadge-stub
        data-qa-selector="dependency_list_total_count"
        pill=""
      >
        100
      </glbadge-stub>
    </h3>
     
    <dependenciesactions-stub
      namespace="allDependencies"
    />
  </div>
   
  <paginateddependenciestable-stub
    namespace="allDependencies"
  />
</div>
`;

exports[`DependenciesApp component on creation given the dependency list job failed matches the snapshot 1`] = `
<div>
  <!---->
   
  <dependencylistjobfailedalert-stub
    jobpath="/jobs/foo/321"
  />
   
  <div
    class="d-sm-flex justify-content-between align-items-baseline my-2"
  >
    <h3
      class="h5"
    >
      
        Dependencies
        
      <!---->
    </h3>
     
    <dependenciesactions-stub
      namespace="allDependencies"
    />
  </div>
   
  <paginateddependenciestable-stub
    namespace="allDependencies"
  />
</div>
`;

exports[`DependenciesApp component on creation given the dependency list job has not yet run matches the snapshot 1`] = `
<glemptystate-stub
  description="The dependency list details information about the components used within your project."
  primarybuttonlink="http://test.host"
  primarybuttontext="Learn more about the dependency list"
  svgpath="/bar.svg"
  title="View dependency details for your project"
/>
`;

exports[`DependenciesApp component on creation matches the snapshot 1`] = `
<glloadingicon-stub
  class="mt-4"
  color="orange"
  label="Loading"
  size="md"
/>
`;
